File: app/src/main/java/com/example/tanawatk/mvvmandroid/di/module/ActivityModule.java
package com.example.tanawatk.mvvmandroid.di.module;

import android.content.Context;

import com.example.tanawatk.mvvmandroid.di.scope.ActivityScope;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {

    private Context context;

    public ActivityModule(Context context) {
        this.context = context;
    }

    @Provides
    @ActivityScope
    public Context provideActivityContext(){
        return context;
    }
}

File: app/src/main/java/com/example/tanawatk/mvvmandroid/di/module/ActivityModule.kt
package com.example.tanawatk.mvvmandroid.di.module

import android.content.Context
import com.example.tanawatk.mvvmandroid.di.scope.ActivityScope
import dagger.Module
import dagger.Provides

@Module
class ActivityModule(private val context: Context) {

    @Provides @ActivityScope
    fun provideActivityContext(): Context = context
}
